### ISSUE TITLE
Extremely distorted and sped-up playback for some seemingly normal FLAC files

### ISSUE THREAD

**Issue by Pheo-Player** [audiocogs/flac.js#15]
I'm having massive problems with playing back FLACs from one specific album. They all are extremely sped up and distorted when played back with aurora/flac.js, but in any other player, they behave perfectly normal. Decoding them and analyzing them works without error, so they are apparently not corrupted in any way.

I'm not sure about the specifics that might cause this, maybe the files are encoded in an unusual manner. The only quirk I could find was in the analysis file, these FLACs often have residual_type=RICE2 on some subframes, while other FLACs I have checked do not. But I don't know too much about this, I just tried to figure out what was off with these specific files.

If necessary, I could possibly send one of the FLACs that evoke this behaviour, or its .ana file.

**Comment by devongovett** [comment]
Yeah would be great if you could upload one of those flac's somewhere so we can debug.

**Comment by Pheo-Player** [comment]
This is one of them: https://dl.dropboxusercontent.com/u/101768390/2._Etnik_-_Nixon.flac

**Comment by devongovett** [comment]
So the reason it doesn't work is because that is a 24 bit file, which we don't yet support.  JS doesn't have 64 bit integers, so we will have to work around this.  I'm taking a look at this now, and will keep you updated.

**Comment by devongovett** [comment]
Should be fixed by aa47cb9b08dc697b65986bd65043a518b8537b8d and deb2f3935cc944b3b9bdf7d4b93578f0f1103ecd.

Turned out not to need 64 bit ints for this particular 24-bit file since it doesn't use LPC frames.  We still need to do something for those files.

The solution to this particular problem was basically the solution suggested by #9.  The buffer size was incorrectly calculated, leading to the increased speed.  Also, the `sampleShift` stuff was really weird and wrong.  Still not sure why libav does that, but it seems to work better without it.  We didn't notice before since most files we tested with were 16 bit, so `sampleShift` was 0, causing no change to the output.
